File: src/db/seed.ts
import { hash } from "bcryptjs";
import { eq } from "drizzle-orm";
import { db } from "@/db";
import {
  applications,
  companies,
  internships,
  letters,
  students,
  users,
} from "@/db/schema";

/* ------------------------------------------------------------------ */
/* Seed — داده‌های نمونه برای سه نقش + فرصت‌ها + درخواست‌ها             */
/* Idempotent: در صورت وجود داده، چیزی تکراری نمی‌سازد                   */
/* ------------------------------------------------------------------ */

const DEMO_PASSWORD_HASH = (async () => await hash("Password@1234", 10))();

async function ensureUser(
  email: string,
  fullName: string,
  role: "student" | "company" | "admin",
  phone: string,
  city: string
) {
  const existing = await db.query.users.findFirst({ where: eq(users.email, email) });
  if (existing) return existing;
  const [row] = await db
    .insert(users)
    .values({
      email,
      fullName,
      role,
      phone,
      city,
      passwordHash: await DEMO_PASSWORD_HASH,
    })
    .returning();
  return row;
}

export async function seedDatabase() {
  const passwordHash = await DEMO_PASSWORD_HASH;

  /* ---------------------------- Admin ---------------------------- */
  const admin = await ensureUser(
    "admin@university.ac.ir",
    "مدیر سامانه دانشگاه",
    "admin",
    "021-88000000",
    "تهران"
  );

  /* -------------------------- Students --------------------------- */
  const saraUser = await ensureUser(
    "sara@student.ac.ir",
    "سارا محمدی",
    "student",
    "0912-1112233",
    "تهران"
  );
  let sara = await db.query.students.findFirst({
    where: eq(students.userId, saraUser.id),
  });
  if (!sara) {
    [sara] = await db
      .insert(students)
      .values({
        userId: saraUser.id,
        university: "دانشگاه صنعتی شریف",
        major: "مهندسی کامپیوتر",
        grade: "کارشناسی",
        studentNumber: "99120001",
        gpa: "17.8",
        skills: ["React", "TypeScript", "Node.js", "SQL"],
        interests: "توسعه فرانت‌اند و طراحی رابط کاربری",
        about:
          "دانشجوی ترم آخر مهندسی کامپیوتر با علاقه به توسعه وب و تجربه ساخت چند پروژه دانشگاهی و شخصی.",
        resumeUrl: "/resumes/sara-mohammadi.pdf",
      })
      .returning();
  }

  const aliUser = await ensureUser(
    "ali@student.ac.ir",
    "علی رضایی",
    "student",
    "0913-5556677",
    "اصفهان"
  );
  let ali = await db.query.students.findFirst({
    where: eq(students.userId, aliUser.id),
  });
  if (!ali) {
    [ali] = await db
      .insert(students)
      .values({
        userId: aliUser.id,
        university: "دانشگاه صنعتی اصفهان",
        major: "مهندسی برق",
        grade: "کارشناسی",
        studentNumber: "98070012",
        gpa: "16.2",
        skills: ["PLC", "مدارهای الکترونیکی", "AutoCAD", "MATLAB"],
        interests: "اتوماسیون صنعتی و سیستم‌های کنترل",
        about: "دانشجوی مهندسی برق قدرت، علاقه‌مند به اتوماسیون صنعتی.",
        resumeUrl: "/resumes/ali-rezaei.pdf",
      })
      .returning();
  }

  const maryamUser = await ensureUser(
    "maryam@student.ac.ir",
    "مریم احمدی",
    "student",
    "0917-8889900",
    "شیراز"
  );
  let maryam = await db.query.students.findFirst({
    where: eq(students.userId, maryamUser.id),
  });
  if (!maryam) {
    [maryam] = await db
      .insert(students)
      .values({
        userId: maryamUser.id,
        university: "دانشگاه شیراز",
        major: "مهندسی صنایع",
        grade: "کارشناسی ارشد",
        studentNumber: "97150034",
        gpa: "18.1",
        skills: ["برنامه‌ریزی تولید", "کنترل کیفیت", "Excel پیشرفته"],
        interests: "بهینه‌سازی فرآیندهای تولید",
        about: "دانشجوی ارشد مهندسی صنایع، فعال در پروژه‌های بهینه‌سازی.",
        resumeUrl: "/resumes/maryam-ahmadi.pdf",
      })
      .returning();
  }

  const hosseinUser = await ensureUser(
    "hossein@student.ac.ir",
    "حسین کریمی",
    "student",
    "0413-4455667",
    "تبریز"
  );
  let hossein = await db.query.students.findFirst({
    where: eq(students.userId, hosseinUser.id),
  });
  if (!hossein) {
    [hossein] = await db
      .insert(students)
      .values({
        userId: hosseinUser.id,
        university: "دانشگاه تبریز",
        major: "مهندسی مکانیک",
        grade: "کارشناسی",
        studentNumber: "99030123",
        gpa: "15.9",
        skills: ["SolidWorks", "ABAQUS", "مکانیک سیالات"],
        interests: "طراحی و تحلیل مکانیکی",
        about: "دانشجوی مکانیک با سابقه کارگاه‌های عملی.",
        resumeUrl: "/resumes/hossein-karimi.pdf",
      })
      .returning();
  }

  /* -------------------------- Companies -------------------------- */
  const parsUser = await ensureUser(
    "hr@parstech.ir",
    "نماینده پارس تکنولوژی",
    "company",
    "021-44001122",
    "تهران"
  );
  let pars = await db.query.companies.findFirst({
    where: eq(companies.userId, parsUser.id),
  });
  if (!pars) {
    [pars] = await db
      .insert(companies)
      .values({
        userId: parsUser.id,
        name: "شرکت پارس تکنولوژی",
        industry: "فناوری اطلاعات",
        description:
          "شرکت نرم‌افزاری فعال در حوزه توسعه وب و اپلیکیشن‌های موبایل با بیش از ۱۰ سال سابقه.",
        address: "تهران، خیابان ولیعصر، برج فناوری، طبقه ۸",
        website: "https://parstech.ir",
        contactPhone: "021-44001122",
        licenseNumber: "IT-1402-8851",
        status: "approved",
      })
      .returning();
  }

  const fooladUser = await ensureUser(
    "hr@foolad.ir",
    "نماینده فولاد کویر",
    "company",
    "031-32223344",
    "اصفهان"
  );
  let foolad = await db.query.companies.findFirst({
    where: eq(companies.userId, fooladUser.id),
  });
  if (!foolad) {
    [foolad] = await db
      .insert(companies)
      .values({
        userId: fooladUser.id,
        name: "کارخانه فولاد کویر",
        industry: "فولاد و متالورژی",
        description:
          "تولیدکننده بزرگ مقاطع فولادی در منطقه مرکزی کشور با دو خط تولید فعال.",
        address: "اصفهان، شهرک صنعتی بزرگ، خیابان فولاد، پلاک ۱۲",
        website: "https://fooladkavir.ir",
        contactPhone: "031-32223344",
        licenseNumber: "IND-1390-2214",
        status: "approved",
      })
      .returning();
  }

  const aryanUser = await ensureUser(
    "hr@aryan.ir",
    "نماینده داده‌پرداز آرین",
    "company",
    "051-37778899",
    "مشهد"
  );
  let aryan = await db.query.companies.findFirst({
    where: eq(companies.userId, aryanUser.id),
  });
  if (!aryan) {
    [aryan] = await db
      .insert(companies)
      .values({
        userId: aryanUser.id,
        name: "گروه داده‌پرداز آرین",
        industry: "فناوری اطلاعات",
        description: "شرکت ارائه‌دهنده راهکارهای داده و هوش تجاری.",
        address: "مشهد، بلوار فناوری، مجتمع نوآوری",
        website: "https://aryandata.ir",
        contactPhone: "051-37778899",
        licenseNumber: "IT-1403-9012",
        status: "pending",
      })
      .returning();
  }

  const barghUser = await ensureUser(
    "hr@bargh.ir",
    "نماینده نیروگاه برق جنوب",
    "company",
    "071-36667788",
    "شیراز"
  );
  let bargh = await db.query.companies.findFirst({
    where: eq(companies.userId, barghUser.id),
  });
  if (!bargh) {
    [bargh] = await db
      .insert(companies)
      .values({
        userId: barghUser.id,
        name: "نیروگاه برق جنوب",
        industry: "انرژی",
        description: "نیروگاه تولید برق منطقه جنوب کشور.",
        address: "شیراز، جاده صنعتی، نیروگاه سیکل ترکیبی",
        website: "https://barghjonoob.ir",
        contactPhone: "071-36667788",
        licenseNumber: "ENE-1385-4471",
        status: "approved",
      })
      .returning();
  }

  /* ------------------------- Internships ------------------------- */
  async function ensureInternship(data: {
    companyId: number;
    title: string;
    description: string;
    capacity: number;
    requiredSkills: string[];
    city: string;
    major: string;
    startDate: string;
    endDate: string;
    conditions: string;
  }) {
    const existing = await db.query.internships.findFirst({
      where: (t, { eq: e, and: a }) =>
        a(e(t.companyId, data.companyId), e(t.title, data.title)),
    });
    if (existing) return existing;
    const [row] = await db.insert(internships).values(data).returning();
    return row;
  }

  const i1 = await ensureInternship({
    companyId: pars.id,
    title: "کارآموز توسعه فرانت‌اند (React)",
    description:
      "همکاری در تیم توسعه محصول برای پیاده‌سازی رابط کاربری پروژه‌های واقعی با React و TypeScript. منتور اختصاصی در طول دوره.",
    capacity: 4,
    requiredSkills: ["React", "TypeScript", "HTML/CSS", "Git"],
    city: "تهران",
    major: "مهندسی کامپیوتر",
    startDate: "2026-03-01",
    endDate: "2026-05-30",
    conditions:
      "حضوری (۳ روز در هفته)، ساعت کاری منعطف ۹ تا ۱۵، کمکهزینه ماهانه ۵ میلیون تومان، امکان جذب پس از پایان دوره.",
  });

  const i2 = await ensureInternship({
    companyId: pars.id,
    title: "کارآموز توسعه Backend (Node.js)",
    description:
      "توسعه و نگهداری سرویس‌های سمت سرور با Node.js و PostgreSQL زیر نظر تیم فنی ارشد.",
    capacity: 2,
    requiredSkills: ["Node.js", "PostgreSQL", "REST API"],
    city: "تهران",
    major: "مهندسی کامپیوتر",
    startDate: "2026-03-15",
    endDate: "2026-06-15",
    conditions:
      "ترکیبی (حضوری/دورکاری)، کمکهزینه ماهانه ۴ میلیون تومان، تیم کوچک و صمیمی.",
  });

  const i3 = await ensureInternship({
    companyId: foolad.id,
    title: "کارآموز اتوماسیون صنعتی",
    description:
      "آشنایی عملی با سیستم‌های PLC و خطوط تولید خودکار، همکاری با واحد نگهداری و تعمیرات.",
    capacity: 3,
    requiredSkills: ["PLC", "مدارهای الکترونیکی", "AutoCAD"],
    city: "اصفهان",
    major: "مهندسی برق",
    startDate: "2026-04-01",
    endDate: "2026-07-01",
    conditions:
      "حضوری، ۸ صبح تا ۱۴، سرویس رفت‌وآمد از شهر، بیمه کارآموزی، کمکهزینه ماهانه ۳.۵ میلیون تومان.",
  });

  const i4 = await ensureInternship({
    companyId: foolad.id,
    title: "کارآموز برنامه‌ریزی تولید",
    description:
      "همکاری با واحد برنامه‌ریزی در زمان‌بندی خطوط تولید و تحلیل داده‌های عملکرد.",
    capacity: 2,
    requiredSkills: ["برنامه‌ریزی تولید", "Excel پیشرفته", "کنترل کیفیت"],
    city: "اصفهان",
    major: "مهندسی صنایع",
    startDate: "2026-04-15",
    endDate: "2026-06-30",
    conditions: "حضوری، ۵ روز در هفته، کمکهزینه ماهانه ۳ میلیون تومان.",
  });

  const i5 = await ensureInternship({
    companyId: bargh.id,
    title: "کارآموز نگهداری و تعمیرات تجهیزات",
    description:
      "همراهی با تیم تعمیرات توربین و تجهیزات نیروگاهی، آشنایی با مستندات فنی.",
    capacity: 2,
    requiredSkills: ["SolidWorks", "مکانیک سیالات", "گزارش‌نویسی فنی"],
    city: "شیراز",
    major: "مهندسی مکانیک",
    startDate: "2026-03-20",
    endDate: "2026-06-20",
    conditions:
      "حضوری، شیفت صبح، بیمه حادثه، کمکهزینه ماهانه ۴ میلیون تومان.",
  });

  const i6 = await ensureInternship({
    companyId: bargh.id,
    title: "کارآموز کنترل کیفیت (QC)",
    description:
      "بازرسی و کنترل کیفیت قطعات و تجهیزات، تهیه گزارش‌های آزمون.",
    capacity: 1,
    requiredSkills: ["کنترل کیفیت", "MATLAB", "Excel پیشرفته"],
    city: "شیراز",
    major: "مهندسی برق",
    startDate: "2026-04-01",
    endDate: "2026-05-30",
    conditions: "حضوری، ۴ روز در هفته، کمکهزینه ماهانه ۳ میلیون تومان.",
  });

  /* ------------------------ Applications ------------------------- */
  async function ensureApplication(
    internshipId: number,
    studentId: number,
    status: "pending" | "accepted" | "rejected",
    coverLetter: string
  ) {
    const existing = await db.query.applications.findFirst({
      where: (t, { eq: e, and: a }) =>
        a(e(t.internshipId, internshipId), e(t.studentId, studentId)),
    });
    if (existing) return existing;
    const [row] = await db
      .insert(applications)
      .values({ internshipId, studentId, status, coverLetter })
      .returning();
    return row;
  }

  await ensureApplication(
    i1.id,
    sara.id,
    "accepted",
    "سلام، من دانشجوی ترم آخر مهندسی کامپیوتر هستم و تجربه عملی در React و TypeScript دارم. بسیار علاقه‌مندم که در تیم پارس تکنولوژی رشد کنم."
  );
  await ensureApplication(
    i2.id,
    sara.id,
    "pending",
    "علاقه‌مند به توسعه سمت سرور هستم و پروژه‌هایی با Node.js انجام داده‌ام."
  );
  await ensureApplication(
    i3.id,
    ali.id,
    "accepted",
    "با PLC و مدارهای الکترونیکی آشنایی عملی دارم و مشتاق یادگیری اتوماسیون در محیط واقعی صنعتی هستم."
  );
  await ensureApplication(
    i4.id,
    maryam.id,
    "pending",
    "پایان‌نامه من در حوزه بهینه‌سازی زمان‌بندی تولید است و می‌توانم به واحد برنامه‌ریزی شما کمک کنم."
  );
  await ensureApplication(
    i5.id,
    hossein.id,
    "rejected",
    "آشنایی کامل با SolidWorks دارم و به نگهداری تجهیزات صنعتی علاقه‌مندم."
  );

  /* --------------------------- Letters ---------------------------- */
  const existingLetter = await db.query.letters.findFirst({
    where: eq(letters.serialNo, "MN-1404-0001"),
  });
  if (!existingLetter) {
    const saraApp = await db.query.applications.findFirst({
      where: (t, { eq: e, and: a }) =>
        a(e(t.internshipId, i1.id), e(t.studentId, sara.id)),
    });
    if (saraApp) {
      await db.insert(letters).values({
        studentId: sara.id,
        internshipId: i1.id,
        applicationId: saraApp.id,
        serialNo: "MN-1404-0001",
      university: sara.university ?? "دانشگاه",
      studentName: saraUser.fullName,
      studentNumber: sara.studentNumber,
      studentMajor: sara.major,
      studentGrade: sara.grade,
      companyName: pars.name,
      internshipTitle: i1.title,
      startDate: i1.startDate ?? "",
      endDate: i1.endDate ?? "",
      });
    }
  }

  return { passwordHash, admin, sara, ali, maryam, hossein };
}

/* ----------------------- Lazy runtime seed ----------------------- */
declare global {
  // eslint-disable-next-line no-var
  var __karamoozyarSeeded: boolean | undefined;
}

let seedingPromise: Promise<void> | null = null;

export async function ensureSeed() {
  if (process.env.NEXT_PHASE === "phase-production-build") return;
  if (globalThis.__karamoozyarSeeded) return;
  if (!seedingPromise) {
    seedingPromise = seedDatabase()
      .then(() => {
        globalThis.__karamoozyarSeeded = true;
      })
      .catch((err) => {
        console.error("Seed failed:", err);
        seedingPromise = null;
      });
  }
  await seedingPromise;
}

File: backend/accounts/management/commands/seed.py
"""
Management command: python manage.py seed
Seeds the database with sample data for کارآموزیار
"""

from django.core.management.base import BaseCommand
from django.utils import timezone

from accounts.models import User, Student, Company
from internships.models import Internship, Application, Letter


class Command(BaseCommand):
    help = "Seed database with sample data for کارآموزیار"

    def handle(self, *args, **options):
        self.stdout.write("🌱 Seeding database...")

        # ---- Admin (University) ----
        admin_user, created = User.objects.get_or_create(
            email="admin@university.ac.ir",
            defaults={
                "username": "admin",
                "full_name": "دکتر احمد محمدی",
                "role": User.Role.ADMIN,
                "phone": "021-12345678",
                "city": "تهران",
                "is_staff": True,
                "is_superuser": True,
            },
        )
        if created:
            admin_user.set_password("admin1234")
            admin_user.save()
            self.stdout.write(f"  ✓ Admin: {admin_user.email} (also Django superuser)")
        else:
            self.stdout.write(f"  ⚠ Admin already exists: {admin_user.email}")

        # ---- Students ----
        students_data = [
            {
                "username": "ali",
                "email": "ali@student.ac.ir",
                "full_name": "علی رضایی",
                "phone": "09121234567",
                "city": "تهران",
                "university": "دانشگاه تهران",
                "major": "مهندسی کامپیوتر",
                "grade": "کارشناسی",
                "student_number": "9812001",
                "gpa": "17.5",
                "skills": ["Python", "Django", "React", "PostgreSQL"],
                "interests": "هوش مصنوعی و توسعه وب",
            },
            {
                "username": "sara",
                "email": "sara@student.ac.ir",
                "full_name": "سارا احمدی",
                "phone": "09131234567",
                "city": "اصفهان",
                "university": "دانشگاه صنعتی اصفهان",
                "major": "مهندسی برق",
                "grade": "کارشناسی ارشد",
                "student_number": "9912002",
                "gpa": "18.2",
                "skills": ["MATLAB", "Simulink", "Arduino", "C++"],
                "interests": "رباتیک و اتوماسیون",
            },
            {
                "username": "mehdi",
                "email": "mehdi@student.ac.ir",
                "full_name": "مهدی کریمی",
                "phone": "09141234567",
                "city": "تبریز",
                "university": "دانشگاه تبریز",
                "major": "مهندسی مکانیک",
                "grade": "کارشناسی",
                "student_number": "4001003",
                "gpa": "16.8",
                "skills": ["SolidWorks", "AutoCAD", "CATIA"],
                "interests": "طراحی صنعتی و CAD/CAM",
            },
            {
                "username": "fatemeh",
                "email": "fatemeh@student.ac.ir",
                "full_name": "فاطمه حسینی",
                "phone": "09151234567",
                "city": "مشهد",
                "university": "دانشگاه فردوسی مشهد",
                "major": "مهندسی کامپیوتر",
                "grade": "کارشناسی ارشد",
                "student_number": "9912004",
                "gpa": "18.9",
                "skills": ["Python", "TensorFlow", "Keras", "NLP", "Computer Vision"],
                "interests": "یادگیری عمیق و پردازش تصویر",
            },
        ]

        students = []
        for data in students_data:
            user = User.objects.create_user(
                username=data["username"],
                email=data["email"],
                password="student1234",
                full_name=data["full_name"],
                role=User.Role.STUDENT,
                phone=data["phone"],
                city=data["city"],
            )
            student = Student.objects.create(
                user=user,
                university=data["university"],
                major=data["major"],
                grade=data["grade"],
                student_number=data["student_number"],
                gpa=data["gpa"],
                skills=data["skills"],
                interests=data["interests"],
            )
            students.append(student)
            self.stdout.write(f"  ✓ Student: {user.email}")

        # ---- Companies ----
        companies_data = [
            {
                "username": "digikala",
                "email": "hr@digikala.com",
                "full_name": "محمد نوری",
                "name": "دیجی‌کالا",
                "industry": "تجارت الکترونیک",
                "description": "بزرگ‌ترین فروشگاه اینترنتی ایران",
                "address": "تهران، خیابان ولیعصر",
                "website": "https://www.digikala.com",
                "contact_phone": "021-91000123",
                "license_number": "12345678",
                "status": Company.Status.APPROVED,
            },
            {
                "username": "snapp",
                "email": "hr@snapp.ir",
                "full_name": "زهرا رحیمی",
                "name": "اسنپ",
                "industry": "فناوری اطلاعات",
                "description": "سرویس حمل و نقل هوشمند",
                "address": "تهران، میدان ونک",
                "website": "https://www.snapp.ir",
                "contact_phone": "021-91000456",
                "license_number": "23456789",
                "status": Company.Status.APPROVED,
            },
            {
                "username": "mapna",
                "email": "hr@mapnagroup.com",
                "full_name": "رضا کاظمی",
                "name": "گروه مپنا",
                "industry": "انرژی و صنعت",
                "description": "شرکت مدیریت پروژه‌های نیروگاهی ایران",
                "address": "تهران، شهرک غرب",
                "website": "https://www.mapnagroup.com",
                "contact_phone": "021-88097000",
                "license_number": "34567890",
                "status": Company.Status.APPROVED,
            },
            {
                "username": "kaveh",
                "email": "hr@kavehsteel.com",
                "full_name": "حسن مرادی",
                "name": "فولاد کاوه جنوب",
                "industry": "صنایع فلزی",
                "description": "تولیدکننده فولاد و مقاطع فلزی",
                "address": "بندرعباس، شهرک صنعتی",
                "website": "https://www.kavehsteel.com",
                "contact_phone": "076-33551234",
                "license_number": "45678901",
                "status": Company.Status.PENDING,
            },
        ]

        companies = []
        for data in companies_data:
            user = User.objects.create_user(
                username=data["username"],
                email=data["email"],
                password="company1234",
                full_name=data["full_name"],
                role=User.Role.COMPANY,
                phone=data["contact_phone"],
                city="تهران",
            )
            company = Company.objects.create(
                user=user,
                name=data["name"],
                industry=data["industry"],
                description=data["description"],
                address=data["address"],
                website=data["website"],
                contact_phone=data["contact_phone"],
                license_number=data["license_number"],
                status=data["status"],
            )
            companies.append(company)
            self.stdout.write(f"  ✓ Company: {company.name} ({company.status})")

        # ---- Internships ----
        internships_data = [
            {
                "company": companies[0],
                "title": "کارآموز توسعه بک‌اند (Python/Django)",
                "description": "فرصت کارآموزی ۳ ماهه در تیم بک‌اند دیجی‌کالا با تمرکز بر توسعه API‌های RESTful و کار با Django.",
                "capacity": 3,
                "required_skills": ["Python", "Django", "REST API", "PostgreSQL"],
                "city": "تهران",
                "major": "مهندسی کامپیوتر",
                "start_date": "2026-09-01",
                "end_date": "2026-12-01",
                "conditions": "حضور حداقل ۳ روز در هفته، بیمه تکمیلی، ناهار رایگان",
            },
            {
                "company": companies[0],
                "title": "کارآموز فرانت‌اند (React)",
                "description": "پیوستن به تیم فرانت‌اند برای توسعه رابط کاربری اپلیکیشن‌های وب.",
                "capacity": 2,
                "required_skills": ["React", "JavaScript", "TypeScript", "HTML/CSS"],
                "city": "تهران",
                "major": "مهندسی کامپیوتر",
                "start_date": "2026-09-15",
                "end_date": "2026-12-15",
                "conditions": "دورکاری امکان‌پذیر، حقوق کارآموزی",
            },
            {
                "company": companies[1],
                "title": "کارآموز علم داده و تحلیل",
                "description": "تحلیل داده‌های رفتار کاربران و مدل‌سازی پیش‌بینی تقاضا.",
                "capacity": 2,
                "required_skills": ["Python", "Pandas", "SQL", "Machine Learning"],
                "city": "تهران",
                "major": "مهندسی کامپیوتر",
                "start_date": "2026-10-01",
                "end_date": "2027-01-01",
                "conditions": "حضور ۴ روز در هفته، حقوق رقابتی",
            },
            {
                "company": companies[1],
                "title": "کارآموز مهندسی برق — سیستم‌های الکترونیکی",
                "description": "همکاری در طراحی و تست بردهای الکترونیکی برای سیستم ناوبری.",
                "capacity": 1,
                "required_skills": ["الکترونیک دیجیتال", "PCB Design", "Arduino"],
                "city": "تهران",
                "major": "مهندسی برق",
                "start_date": "2026-09-01",
                "end_date": "2026-12-01",
                "conditions": "تمام‌وقت، حقوق ماهانه",
            },
            {
                "company": companies[2],
                "title": "کارآموز مهندسی مکانیک — طراحی",
                "description": "کارآموزی در واحد طراحی مپنا با کار بر روی پروژه‌های نیروگاهی.",
                "capacity": 4,
                "required_skills": ["SolidWorks", "AutoCAD", "ترمودینامیک"],
                "city": "تهران",
                "major": "مهندسی مکانیک",
                "start_date": "2026-09-01",
                "end_date": "2027-02-01",
                "conditions": "بیمه، ناهار، سرویس رفت و آمد",
            },
        ]

        internships = []
        for data in internships_data:
            internship = Internship.objects.create(
                company=data["company"],
                title=data["title"],
                description=data["description"],
                capacity=data["capacity"],
                required_skills=data["required_skills"],
                city=data["city"],
                major=data["major"],
                start_date=data["start_date"],
                end_date=data["end_date"],
                conditions=data["conditions"],
            )
            internships.append(internship)
            self.stdout.write(f"  ✓ Internship: {internship.title}")

        # ---- Applications ----
        applications_data = [
            {"student": students[0], "internship": internships[0], "status": Application.Status.ACCEPTED, "cover_letter": "علاقه‌مند به یادگیری Django و توسعه بک‌اند هستم."},
            {"student": students[0], "internship": internships[2], "status": Application.Status.PENDING, "cover_letter": "تجربه کار با Python و Pandas دارم."},
            {"student": students[1], "internship": internships[3], "status": Application.Status.ACCEPTED, "cover_letter": "در زمینه الکترونیک دیجیتال تجربه عملی دارم."},
            {"student": students[2], "internship": internships[4], "status": Application.Status.ACCEPTED, "cover_letter": "علاقه‌مند به طراحی صنعتی هستم و با SolidWorks کار کرده‌ام."},
            {"student": students[3], "internship": internships[0], "status": Application.Status.PENDING, "cover_letter": "تجربه با Django و REST API دارم."},
            {"student": students[3], "internship": internships[2], "status": Application.Status.REJECTED, "cover_letter": "مهارت‌های ML و تحلیل داده دارم."},
        ]

        applications = []
        for data in applications_data:
            app = Application.objects.create(
                student=data["student"],
                internship=data["internship"],
                status=data["status"],
                cover_letter=data["cover_letter"],
            )
            applications.append(app)
            self.stdout.write(f"  ✓ Application: {app.student.user.full_name} → {app.internship.title} ({app.status})")

        # ---- Letters (for accepted applications) ----
        accepted_apps = [a for a in applications if a.status == Application.Status.ACCEPTED]
        for i, app in enumerate(accepted_apps):
            serial = f"MN-1404-{str(i + 1).zfill(4)}"
            Letter.objects.create(
                student=app.student,
                internship=app.internship,
                application=app,
                serial_no=serial,
                university=app.student.university or "دانشگاه",
                student_name=app.student.user.full_name,
                student_number=app.student.student_number or "",
                student_major=app.student.major or "",
                student_grade=app.student.grade or "",
                company_name=app.internship.company.name,
                internship_title=app.internship.title,
                start_date=str(app.internship.start_date or ""),
                end_date=str(app.internship.end_date or ""),
            )
            self.stdout.write(f"  ✓ Letter: {serial}")

        self.stdout.write(self.style.SUCCESS("\n🎉 Database seeded successfully!"))
        self.stdout.write("\n📋 Test Accounts:")
        self.stdout.write("  Admin:    admin@university.ac.ir / admin1234")
        self.stdout.write("  Students: ali@student.ac.ir / student1234")
        self.stdout.write("            sara@student.ac.ir / student1234")
        self.stdout.write("  Companies: hr@digikala.com / company1234")
        self.stdout.write("             hr@snapp.ir / company1234")
